pantheon/src/main/java/tech/pegasys/pantheon/cli/PantheonCommand.java

```diff
@@ -1067,10 +1067,16 @@ private EthNetworkConfig updateNetworkConfig(final NetworkName network) {
               genesisConfigFile
                   .getConfigOptions()
                   .getChainId()
+                  .map(BigInteger::intValueExact)
                   .orElse(EthNetworkConfig.getNetworkConfig(MAINNET).getNetworkId()));
         } catch (final DecodeException e) {
           throw new ParameterException(
               this.commandLine, String.format("Unable to parse genesis file %s.", genesisFile), e);
+        } catch (final ArithmeticException e) {
+          throw new ParameterException(
+              this.commandLine,
+              "No networkId specified and chainId in "
+                  + "genesis file is too large to be used as a networkId");
```

Comment:
ugh, this is quite unfortunate but probably better to bail out with an error than to default to something else silently.

Reply:
yeah it's not nice, but I think it's the best thing to do